homeassistant/components/mqtt/client.py

```diff
@@ -907,11 +910,13 @@ async def _discovery_cooldown(self) -> None:
         wait_until = max(
             last_discovery + DISCOVERY_COOLDOWN, last_subscribe + DISCOVERY_COOLDOWN
         )
-        while now < wait_until:
+        while now < wait_until or self._pending_subscriptions:
             await asyncio.sleep(wait_until - now)
```

Comment:
It should not be possible that wait_until < now, but since the while condition is changed, it's no longer necessarily guaranteed  make this a `min(wait_until - now, 0)` maybe?

Reply:
This is current code, not going to change this.